peps/pep-0813.rst

```diff
@@ -0,0 +1,196 @@
+made even more convenient, since no extra imports are required, and users can easily just piggyback on
+well-worn "print debugging" patterns, at least for the most common use cases.
+
+These two extensions work independently, but hand-in-hand can provide a powerful and convenient new feature.
+
+
+Specification
+=============
+
+There are two parts to this proposal.
+
+
+``__pretty__()`` methods
+------------------------
+
+Classes can implement a new dunder method, ``__pretty__()`` which if present, generates the pretty printed
+representation of their instances.  This augments ``__repr__()`` which, prior to this proposal, was the only
+method used to generate a pretty representation of the object.  Since object reprs provide functionality
+distinct from pretty printing, some classes may want more control over their pretty display.
+
+``__pretty__()`` is optional; if missing, the standard pretty printers fall back to ``__repr__()``
+for full backward compatibility (technically speaking, :py:func:`python:pprint.saferepr` is used).
+However, if defined on a class, ``__pretty__()`` has the same argument signature as
+:py:meth:`python:pprint.PrettyPrinter.format`, taking four arguments:
+
+* ``object`` - the object to print, which is effectively always ``self``
+* ``context`` - a dictionary mapping the ``id()`` of objects which are part of the current presentation
+  context
+* ``maxlevels`` - the requested limit to recursion
+* ``levels`` - the current recursion level
+
+Similarly, ``__pretty__()`` returns three values, the string to be used as the pretty printed representation,
+a boolean indicating whether the returned value is "readable", and a boolean indicating whether recursion has
+been detected.  In this context, "readable" means the same as
+:py:meth:`python:pprint.PrettyPrinter.isreadable`, i.e. that the returned value can be used to reconstruct the
+original object using ``eval()``.
+
+See :py:meth:`python:pprint.PrettyPrinter.format` for details.
+
+
+A new argument to built-in ``print``
+------------------------------------
+
+Built-in :func:`print` takes a new optional argument, appended to the end of the argument list, called
+``pretty``, which can take one of the following values:
+
+* ``None`` - the default. No pretty printing is invoked.  Fully backward compatible.
+* ``True`` - use a temporary instance of the :py:class:`python:pprint.PrettyPrinter` class to get a
+  pretty representation of the object.
+* An instance with a ``pformat()`` method, which has the same signature as
+  :py:meth:`python:pprint.PrettyPrinter.pformat`.  When given, this will usually be an instance of a
+  subclass of ``PrettyPrinter`` with its ``pformat()`` method overridden.  Note that this form
+  requires **an instance** of a pretty printer, not a class, as only ``print(..., pretty=True)``
+  performs implicit instantiation.
+
+
+Examples
+========
+
+A custom ``__pprint__()`` method can be used to customize the representation of the object:
```

Comment:
```suggestion
A custom ``__pretty__()`` method can be used to customize the representation of the object:
```

Reply:
Actually, the dunder _is_ ``__pprint__`` for now.